datumaro/components/merge/intersect_merge.py

```diff
@@ -52,11 +52,68 @@
 
 @attrs
 class IntersectMerge(Merger):
+    """
+    Merge several datasets with "intersect" policy:
+
+    - If there are two or more dataset items whose (id, subset) pairs match each other,
+    we can consider this as having an intersection in our dataset. This method merges
+    the annotations of the corresponding :class:`DatasetItem` into one :class:`DatasetItem`
+    to handle this intersection. The rule to handle merging annotations is provided by
+    :class:`AnnotationMerger` according to their annotation types. For example,
+    DatasetItem(id="item_1", subset="train", annotations=[Bbox(0, 0, 1, 1)]) from Dataset-A and
+    DatasetItem(id="item_1", subset="train", annotations=[Bbox(.5, .5, 1, 1)]) from Dataset-B can be
+    merged into DatasetItem(id="item_1", subset="train", annotations=[Bbox(0, 0, 1, 1)]).
```

Comment:
1. don't we need to consider about media itself? item_1 from Dataset-A and item_1 from Dataset-B could be totally different.
2. I don't understand why merging Bbox(0, 0, 1, 1) and Bbox(.5, .5, 1, 1) is going to be Bbox(0, 0, 1, 1) in an intersection viewpoint.
3. In the similar vein, merging {"car", "bus", "truck"} and {"car", "cat", "dog"} is {"bust", "car", "cat", "dog", "truck"} in an intersection viewpoint.
These are more like be union to me.

Reply:
1. From 
https://github.com/openvinotoolkit/datumaro/blob/c2ff84bdb33bd381d52750ee6d89ae73f59aeb00/datumaro/components/merge/intersect_merge.py#L275-L290

It matches items to be merged only with `(id, subset)` pair.

2. From
https://github.com/openvinotoolkit/datumaro/blob/c2ff84bdb33bd381d52750ee6d89ae73f59aeb00/datumaro/components/annotations/merger.py#L108-L112

It tries to pick the one closest to the cluster (`cluster` is `[Bbox(0, 0, 1, 1), Bbox(.5, .5, 1, 1)]` in this case).

3. From
https://github.com/openvinotoolkit/datumaro/blob/c2ff84bdb33bd381d52750ee6d89ae73f59aeb00/datumaro/components/merge/intersect_merge.py#L292-L334

As we discussed before, `IntersectMerge` exactly works the same with `UnionMerge` in merging label categories. However, I think the spirit under `IntersectMerge` is that when there are **two items matching in both datasets, they are merged into one**. For example, in this case

|        | Item slot 1                                                                              | Item slot 2                                                                                |
|--------|------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------|
| Before | (Dataset-A): `DatasetItem(id="item_1", subset="train", annotations=[Bbox(0, 0, 1, 1)])` | (Dataset-B): `DatasetItem(id="item_1", subset="train", annotations=[Bbox(.5, .5, 1, 1)])` |
|  After | (Merged) `DatasetItem(id="item_1", subset="train", annotations=[Bbox(0, 0, 1, 1)])`        |                                                                                            |

It clearly shows that the number of items: 2 -> 1.

Therefore, the opposite spirit can be given to UnionMerge as follows.

|        | Item slot 1                                                                              | Item slot 2                                                                                |
|--------|------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------|
| Before | (Dataset-A): `DatasetItem(id="item_1", subset="train", annotations=[Bbox(0, 0, 1, 1)])` | (Dataset-B): `DatasetItem(id="item_1", subset="train", annotations=[Bbox(.5, .5, 1, 1)])` |
|  After | (Merged) `DatasetItem(id="item_1-0", subset="train", annotations=[Bbox(0, 0, 1, 1)])`      | (Merged) `DatasetItem(id="item_1-1", subset="train", annotations=[Bbox(.5, .5, 1, 1)])`        |

It shows no reduction in the number of items before and after.